### PR TITLE
create py-circuit-build package

### PR DESCRIPTION
**Pull request by asanin-epfl**
cc @mgeplf

### CODE REVIEW

**Review by matz-e — COMMENTED** [review]
Generic comment… we have `snakemake` and `spykfunc` just like that without `py-`… since `py-circuit-build` is (potentially) more of an end-user package, maybe wwe could just call it `circuit-build`?

Should have said the same about `brain-builder`… unless that's a Python dependency somewhere…

**Comment by pramodk** [comment]
just remark : circuit-build seems very generic name. Should this be something like circuit-build-tools ?

**Comment by matz-e** [comment]
> just remark : circuit-build seems very generic name. Should this be something like circuit-build-tools ?

I'm more in favor of something close to the repository name.

**Comment by pramodk** [comment]
Are you suggesting to change repo name then? 😀

I don’t have strong opinion here but when I saw circuit-build, whole circuit building toolchain came to my mind.

**Comment by matz-e** [comment]
> I don’t have strong opinion here but when I saw circuit-build, whole circuit building toolchain came to my mind.

IIRC, this does just that :)

**Comment by mgeplf** [comment]
> just remark : circuit-build seems very generic name. Should this be something like circuit-build-tools ?

This isn't tools, though, this is the actual `circuit-build` command orchestrates all the tools.

**Comment by pramodk** [comment]
Oh ok, then it's fine 👍

**Comment by asanin-epfl** [comment]
what name to prefer then? `circuit-build` or `py-circuit-build`?

**Comment by matz-e** [comment]
> what name to prefer then? `circuit-build` or `py-circuit-build`?

If this is not used as a dependency elsewhere to load a Python package, I would drop the `py-`.

**Comment by asanin-epfl** [comment]
@matz-e  please review

**Review by matz-e — APPROVED** [review]

File: var/spack/repos/builtin/packages/circuit-build/package.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    homepage = "https://bbpcode.epfl.ch/code/#/admin/projects/common/circuit-build"
+    git      = "ssh://bbpcode.epfl.ch/common/circuit-build"
+
+    version('develop', branch='master')
+    version('3.1.1', tag='circuit-build-v3.1.1')
+
+    depends_on('py-setuptools', type=('build', 'run'))
+
+    depends_on('py-click@7.0:7.99', type='run')
+    depends_on('snakemake@5.6:5.99', type='run')
```
Comment (asanin-epfl) [inline var/spack/repos/builtin/packages/circuit-build/package.py:21]: @matz-e can you please advice. This 'snakemake@5.6:5.99' dependency provides an executable command `snakemake` which does not become available after the module `circuit-build` is loaded. Should I add 'snakemake' to `deploy/configs/applications/modules.yaml` and `deploy/packages/bbp-packages.yaml` in order to provide `snakemake` executable after the module of 'circuit-build' has been loaded?
```
@bbpv1 ~$ module load unstable circuit-build
Autoloading python/3.7.4
@bbpv1 ~$ pip list
Package             Version  
------------------- ---------
...
snakemake           5.10.0   
...
@bbpv1 ~$ snakemake
command not found: snakemake
```

Comment (matz-e) [reply var/spack/repos/builtin/packages/circuit-build/package.py:21]: There's already a snakemake module, don't do that.

You can explicitly request `snakemake`'s `bin` in the path by adding something like
```
def setup_run_environment(self, env):
    env.prepend_path('PATH', self.spec['snakemake'].prefix.bin)
```
grep through the sources to find examples about the exact syntax

**Pull request merged**